mlir/test/mlir-tblgen/op-decl-and-defs.td

```diff
@@ -235,14 +235,14 @@ def NS_FOp : NS_Op<"op_with_all_types_constraint",
 
 // DEFS: FOp FOp::create(::mlir::OpBuilder &builder, ::mlir::Location location, ::mlir::Value a) {
 // DEFS:   ::mlir::OperationState __state__(location, getOperationName());
-// DEFS:   build(builder, __state__, a);
+// DEFS:   build(builder, __state__, static_cast<decltype(a)>(a));
```

Comment:
Wouldn't std::forward also work here? I think that's more idiomatic

Reply:
I think it should be `std::forward<dectlype(foo)>` (which will become a static cast under the hood, but IMO the intent will be more obvious)